misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java

```diff
@@ -274,61 +303,210 @@ public static Symbol findSymbol(List<TypeSymbol> fileContentSymbols) {
      * @param context
      * @param foundTypeLeft  {@link String}
      * @param foundTypeRight {@link String}
+     * @param syntaxTree
+     * @param semanticModel
      * @return function string with mapped schemas
      * @throws IOException throws if error occurred when getting mapped function
      */
     private static String getGeneratedRecordMapping(CodeActionContext context, String foundTypeLeft,
                                                     String foundTypeRight,
-                                                    Symbol lftTypeSymbol, Symbol rhsTypeSymbol)
+                                                    Symbol lftTypeSymbol, Symbol rhsTypeSymbol, SyntaxTree syntaxTree, SemanticModel semanticModel)
 
             throws IOException {
         JsonObject rightRecordJSON = new JsonObject();
         JsonObject leftRecordJSON = new JsonObject();
 
-        // Schema 1
-        TypeSymbol rhsSymbol = ((TypeReferenceTypeSymbol) rhsTypeSymbol).typeDescriptor();
-        if ("RECORD".equals(rhsSymbol.typeKind().name())) {
-            RecordTypeSymbol rightSymbol = (RecordTypeSymbol) rhsSymbol;
-            Map<String, RecordFieldSymbol> rightSchemaFields = rightSymbol.fieldDescriptors();
+        List<RecordTypeSymbol> symbolList = checkMappingCapability(lftTypeSymbol, rhsTypeSymbol);
+        if (!symbolList.contains(null)) {
+            // Schema 1
+
+            // To get the rest field details
+            DataMapperNodeVisitor nodeVisitor = new DataMapperNodeVisitor(foundTypeRight);
+            nodeVisitor.setModel(semanticModel);
+            syntaxTree.rootNode().accept(nodeVisitor);
+            restFieldMap = nodeVisitor.restFields;
+            spreadFieldMap = nodeVisitor.spreadFields;
+
+
+            Map<String, RecordFieldSymbol> rightSchemaFields = symbolList.get(0).fieldDescriptors();
             JsonObject rightSchema = (JsonObject) recordToJSON(rightSchemaFields.values());
 
+            if (!restFieldMap.isEmpty()) {
+                rightSchema = insertRestFields(rightSchema, foundTypeRight);
+            }
+
+            if(!spreadFieldMap.isEmpty()){
```

Comment:
I think we don't need the empty check here.

Reply:
Some times spread fields can be empty.